pkg/metrics/smart/metrics.go

```diff
@@ -110,7 +175,99 @@ func (m *Metrics) WithBlockDeviceCurrentTemperatureValid() *Metrics {
 		},
-		[]string{"blockdevicename", "path", "hostname", "nodename"},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceRotationalLatency declares the metric rotational latency
+// as a prometheus metric
+func (m *Metrics) WithBlockDeviceRotationalLatency() *Metrics {
+	m.blockDeviceRotationalLatency = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_rotational_latency",
+			Help:      `Value of the latency in seconds. 0 indicates it is a SSD, non-zero value indicates a HDD`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceCapacity declares the blockdevice capacity
+func (m *Metrics) WithBlockDeviceCapacity() *Metrics {
+	m.blockDeviceCapacity = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_capacity",
+			Help:      `Capacity of the block device in bytes`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceRotationRate declares the rotational rate of a block device
+func (m *Metrics) WithBlockDeviceRotationRate() *Metrics {
+	m.blockDeviceRotationRate = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_rotational_rate",
+			Help:      `Rotational rate of the block device in RPM`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceTotalBytesRead declares the total number of bytes read by a block device
+func (m *Metrics) WithBlockDeviceTotalBytesRead() *Metrics {
+	m.blockDeviceTotalReadBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_total_read_bytes",
+			Help:      `total number of bytes read by a block device in bytes `,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceTotalBytesWritten declares the total number of bytes written by a block device
+func (m *Metrics) WithBlockDeviceTotalBytesWritten() *Metrics {
+	m.blockDeviceTotalWrittenBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_total_written_bytes",
+			Help:      `total number of bytes written by a block device in bytes `,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceUtilizationRate declares the utilization rate of a block device
+func (m *Metrics) WithBlockDeviceUtilizationRate() *Metrics {
+	m.blockDeviceUtilizationRate = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_utilization_rate",
+			Help:      `utilization rate of a block device `,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDevicePercentEnduranceUsed declares the percentage of endurance used by a block device
+func (m *Metrics) WithBlockDevicePercentEnduranceUsed() *Metrics {
+	m.blockDevicePercentEnduranceUsed = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_endurance_used",
+			Help:      `Percent endurance used of a block device `,
```

Comment:
A better description is "estimate of the percentage of the device life that has been used"